src/inmanta/server/services/environmentservice.py

```diff
@@ -54,16 +56,30 @@ def rename_fields(env: model.Environment) -> JsonType:
     return env_dict
 
 
+class EnvironmentAction(str, Enum):
+    created = "create"
+    deleted = "delete"
+    cleared = "clear"
+    updated = "update"
+
+
+class EnvironmentListener:
+    async def environment_action(self, action: EnvironmentAction, env: model.Environment) -> None:
```

Comment:
Maybe we should add an argument `**kwargs`.  We can then use it to pass on extra action-specific data. This way we can for example pass the name of the setting that has been updated.

Reply:
In that case I would add a method for every action, to preserve typing.